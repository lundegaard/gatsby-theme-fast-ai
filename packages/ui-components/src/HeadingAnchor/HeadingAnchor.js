import React, { forwardRef, useState } from 'react';
import PropTypes from 'prop-types';

import Heading from '../Heading';
import IconButton from '../IconButton';
import IconInsertLink from '../IconInsertLink';

const Anchor = forwardRef(({ to }, ref) => (
	<IconButton
		as="a"
		ref={ref}
		href={to}
		size={22}
		sx={{
			m: 0,
			p: 0,
			ml: 2,
			verticalAlign: 'middle',
			cursor: 'pointer',
		}}
	>
		<IconInsertLink size={22} />
	</IconButton>
));
Anchor.displayName = 'Anchor';
Anchor.propTypes = {
	to: PropTypes.string,
};
const getLink = (href) => {
	if (typeof window === 'undefined') {
		return '';
	}
	const { origin, search, pathname } = document.location;
	return `${origin + pathname + search}#${href}`;
};
const HeadingAnchor = forwardRef(({ children, id, copyLink, ...rest }, ref) => {
	const [inHover, setHover] = useState(false);
	const to = copyLink || getLink(id);

	return (
		<Heading
			onMouseEnter={() => setHover(true)}
			onMouseLeave={() => setHover(false)}
			id={id}
			ref={ref}
			{...rest}
		>
			{children}
			{(id || copyLink) && inHover && <Anchor to={to} />}
		</Heading>
	);
});
HeadingAnchor.displayName = 'HeadingAnchor';

HeadingAnchor.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** You can override the whole link. If not set the current location with
	 * `id` as the hash is used. */
	copyLink: PropTypes.string,
	id: PropTypes.string,
};

export default HeadingAnchor;
