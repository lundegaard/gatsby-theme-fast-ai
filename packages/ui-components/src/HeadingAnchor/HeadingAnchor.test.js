import React from 'react';
import { shallow } from 'enzyme';

import HeadingAnchor from './HeadingAnchor';

describe('<HeadingAnchor />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<HeadingAnchor {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
