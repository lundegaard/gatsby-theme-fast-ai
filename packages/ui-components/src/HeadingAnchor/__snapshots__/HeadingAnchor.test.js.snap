// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadingAnchor /> matches snapshot when shallowly rendered 1`] = `
<Heading
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
